src/OFS/ObjectManager.py

```diff
@@ -589,17 +589,20 @@ def manage_exportObject(self, id='', download=None,
         suffix = 'zexp'
 
         if download:
-            f = BytesIO()
-            ob._p_jar.exportFile(ob._p_oid, f)
+            with BytesIO() as f:
+                ob._p_jar.exportFile(ob._p_oid, f)
+                result = f.getvalue()
+
             if RESPONSE is not None:
                 RESPONSE.setHeader('Content-type', 'application/data')
                 RESPONSE.setHeader('Content-Disposition',
                                    'inline;filename=%s.%s' % (id, suffix))
-            return f.getvalue()
+            return result
```

Comment:
Maybe, I only tried for a couple minutes and couldn't get it to work. Either the file was already closed by the time it reached the publisher or it wasn't being closed at all.